repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -116,25 +116,17 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
       lazy val requestHandler = new KeystoneRequestHandler(keystoneV2Config.getIdentityService.getUri, akkaServiceClient, traceId)
       lazy val isSelfValidating = Option(config.getIdentityService.getUsername).isEmpty ||
         Option(config.getIdentityService.getPassword).isEmpty
-      lazy val tenantFromUriOption: Option[String] = {
-        def extractTenantIdFromUri(uri: String, regexes: Seq[Regex]): Option[String] = {
-          if (regexes.isEmpty) {
-            throw UnparseableTenantException("Could not parse tenant from the URI")
-          } else {
-            val regex = regexes.head
-            uri match {
-              case regex(tid, _*) => Some(tid)
-              case _ => extractTenantIdFromUri(uri, regexes.tail)
-            }
-          }
-        }
-
-        Option(config.getTenantHandling.getValidateTenant) flatMap { validateTenantConfig =>
-          Option(validateTenantConfig.getUriExtractionRegex) flatMap { uriExtractionRegexList =>
-            extractTenantIdFromUri(request.getRequestURI, uriExtractionRegexList.asScala.map(_.r))
-          }
-        }
-      }
+      lazy val tenantFromUriOption: Option[String] =
+        Option(config.getTenantHandling.getValidateTenant).flatMap({ validateTenantConfig =>
+          Option(validateTenantConfig.getUriExtractionRegex).flatMap({ uriExtractionRegexList =>
+            uriExtractionRegexList.asScala.toStream.map(_.r).flatMap({ uriExtractionRegex: Regex =>
+                          request.getRequestURI match {
+                            case uriExtractionRegex(tenantId, _*) => Option(tenantId)
+                            case _ => None
+                          }
+                        }).headOption
```

Comment:
I do have to ask what's up with all of the `({` and `})`, though. In general, you shouldn't need to double up delimiters like that, and I think that holds true here, as well. Also, there is a bunch of whitespace before the pattern match, but it doesn't seem like that whitespace was used to make it line up with anything... so I'm wondering what that's about.

Personally, I would also remove the `Regex` type ascription in the `flatMap` since, considering the variable name, it doesn't really provide additional clarity or type safety. So I propose the following cleanup:

```
lazy val tenantFromUri: String =
  Option(config.getTenantHandling.getValidateTenant).flatMap(validateTenantConfig =>
    Option(validateTenantConfig.getUriExtractionRegex).flatMap(uriExtractionRegexList =>
      uriExtractionRegexList.asScala.toStream.map(_.r).flatMap(uriExtractionRegex =>
        request.getRequestURI match {
          case uriExtractionRegex(tenantId, _*) => Option(tenantId)
          case _ => None
        }
      ).headOption
    )
  ).getOrElse(throw UnparseableTenantException("Could not parse tenant from the URI"))
```